### PR TITLE
tests/lib/fakestore/store:  teach the fake store to fake the new install/refresh endpoint

### PR DESCRIPTION
**Pull request by pedronis**
https://github.com/snapcore/snapd/pull/4772/commits/af4860424f82cd7039e5a919f329d97c6baae4e2 is the interesting commit.

### CODE REVIEW

File: tests/lib/fakestore/store/store.go [hunk 1]
```diff
@@ -482,6 +484,153 @@ func (s *Store) collectAssertions() (asserts.Backstore, error) {
...
+	// check if we have downloadable snap of the given SnapID or name
+	for _, a := range actions {
+		name := a.Name
+		snapID := a.SnapID
+		if a.Action == "refresh" {
+			name = snapIDtoName[snapID]
+		}
+
+		if name == "" {
+			http.Error(w, fmt.Sprintf("unknown snapid: %q", snapID), 400)
```
Comment (zyga) [inline tests/lib/fakestore/store/store.go]: Nitpick, `unknown snap-id: %q`

File: tests/lib/fakestore/store/store.go [hunk 2]
```diff
@@ -482,6 +484,153 @@ func (s *Store) collectAssertions() (asserts.Backstore, error) {
...
+			return
+		}
+
+		if fn, ok := snaps[name]; ok {
+			essInfo, err := snapEssentialInfo(w, fn, snapID, bs)
+			if essInfo == nil {
+				if err != errInfo {
```
Comment (zyga) [inline tests/lib/fakestore/store/store.go]: What is `errInfo` here? Maybe add a comment if you deem it necessary.

File: tests/lib/fakestore/store/store.go [hunk 3]
```diff
@@ -482,6 +484,153 @@ func (s *Store) collectAssertions() (asserts.Backstore, error) {
...
+					Architectures: []string{"all"},
+					SnapID:        essInfo.SnapID,
+					Name:          essInfo.Name,
+					Version:       essInfo.Version,
+					Revision:      essInfo.Revision,
+				},
+			}
+			res.Snap.Publisher.ID = essInfo.DeveloperID
+			res.Snap.Publisher.Username = essInfo.DevelName
+			res.Snap.Download.URL = fmt.Sprintf("%s/download/%s", s.URL(), filepath.Base(fn))
```
Comment (zyga) [inline tests/lib/fakestore/store/store.go]: Totally nitpicky: `path.Join(s.URL(), "download", filepath.Base(fn))`

File: tests/lib/fakestore/store/store.go [hunk 4]
```diff
@@ -482,6 +484,153 @@ func (s *Store) collectAssertions() (asserts.Backstore, error) {
...
+			res.Snap.Download.Size = essInfo.Size
+			replyData.Results = append(replyData.Results, res)
+		}
+	}
+
+	// use indent because this is a development tool, output
+	// should look nice
+	out, err := json.MarshalIndent(replyData, "", "    ")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("can marshal: %v: %v", replyData, err), 400)
```
Comment (zyga) [inline tests/lib/fakestore/store/store.go]: Did you mean `can` or should this be `cannot`?

Comment (pedronis) [reply tests/lib/fakestore/store/store.go]: that's copied code, but seems the problem can vs cannot is widespread here, anyway fixing

File: tests/lib/fakestore/store/store.go (hunk 2 above)
Comment (pedronis) [reply tests/lib/fakestore/store/store.go]: it's a strange pattern in the code here,  it's a designated error but is also the only error that snapEssentialInfo can produce afaict now so the check here is a bit strange, OTOH I'm not keen to change this here now.

**Comment by pedronis** [comment]
this has been merged into #4900 and just leaving that one open

**Pull request closed without merging**